### PR TITLE
Blog validation

### PR DESCRIPTION
**Pull request by Haadiiii**
## BLOG-APP Validation :arrow_double_down: 

In this Activiy: 

> VALIDATION

 Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.

 > SPEC

- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/modals/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  it 'is not valid with a comments_counter being less than 0' do
+    subject.comments_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'is not valid with a likes_counter not being an integer' do
+    subject.likes_counter = 'a'
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (danmainah) [inline spec/modals/post_spec.rb:53]: - [ ] @Haadiiii great work on your tests, however, you forgot to add tests for your custom methods, also I dint see test for like and comments models.

File: spec/modals/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  it 'post counter should be greater than 0' do
+    subject.posts_counter = 1
+    expect(subject).to be_valid
+  end
+
+  it 'post counter should be an integer' do
+    subject.posts_counter = 1.2
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (danmainah) [inline spec/modals/user_spec.rb:27]: - [ ] @Haadiiii great work on your tests, however, you forgot to add tests for your custom methods, also I didnt see tests for like and comments models.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Haadiiii , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/modals/post_spec.rb (hunk 1 above)
Comment (Haadiiii) [reply spec/modals/post_spec.rb:64]: done :+1:

File: spec/modals/user_spec.rb (hunk 2 above)
Comment (Haadiiii) [reply spec/modals/user_spec.rb:38]: done :+1:

**Review by DammyShittu — APPROVED** [review]
Hi @Haadiiii,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**